src/image/loading_displaying.js

```diff
@@ -280,6 +280,79 @@ function _createGif(
   finishCallback();
 }
 
+/**
+ * @private
+ * @param {Constant} xAlign either LEFT, RIGHT or CENTER
+ * @param {Constant} yAlign either TOP, BOTTOM or CENTER
+ * @param {Number} dx
+ * @param {Number} dy
+ * @param {Number} dw
+ * @param {Number} dh
+ * @param {Number} sw
+ * @param {Number} sh
+ * @returns {Object}
+ */
+
+function _imageContain(xAlign, yAlign, dx, dy, dw, dh, sw, sh) {
+  const r = Math.max(sw / dw, sh / dh, 1);
+  const [adjusted_dw, adjusted_dh] = [sw / r, sh / r];
+
+  let x = dx;
+  let y = dy;
+
+  if (xAlign === constants.CENTER) {
+    x = (dw - adjusted_dw) / 2;
+  } else if (xAlign === constants.RIGHT) {
+    x = dw - adjusted_dw;
+  }
+
+  if (yAlign === constants.CENTER) {
+    y = (dh - adjusted_dh) / 2;
+  } else if (yAlign === constants.BOTTOM) {
+    y = dh - adjusted_dh;
+  }
+  return { x, y, w: adjusted_dw, h: adjusted_dh };
+}
+/**
+ * @private
+ * @param {p5.Image|p5.Element|p5.Texture} img the image to draw
+ * @param {Constant} xAlign either LEFT, RIGHT or CENTER
+ * @param {Constant} yAlign either TOP, BOTTOM or CENTER
+ * @param {Number} dw
+ * @param {Number} dh
+ * @param {Number} sx
+ * @param {Number} sy
+ * @param {Number} sw
+ * @param {Number} sh
+ * @returns {Object}
+ */
+
+function _imageCover(img, xAlign, yAlign, dw, dh, sx, sy, sw, sh) {
+  const r = Math.max(dw / sw, dh / sh, 1);
+  img.resize(img.width * r, img.height * r);
```

Comment:
I think we aren't able to use `img.resize` in this method because not all input types have this method (e.g. `p5.MediaElement`, the return type of `createVideo`). Also, `resize` modifies the image object, and I think we want to keep these methods from modifying the input.